src/Storages/MergeTree/MergeTreeReaderWide.cpp

```diff
@@ -180,12 +180,14 @@ void MergeTreeReaderWide::addStreams(const NameAndTypePair & name_and_type,
         if (!data_file_exists)
             return;
 
+        bool is_lc_dict = substream_path.size() > 1 && substream_path[substream_path.size() - 2].type == ISerialization::Substream::Type::DictionaryKeys;
```

Comment:
Maybe it would be better to pass substream_path to MergeTreeReaderStream and it will decide what to do with this info.